### PR TITLE
Allow tile metadata service to generate tilejson for scanned map with single raster

### PR DESCRIPTION
**Pull request by eliotjordan**
Works towards #5399 

Screen Shot 2022-11-15 at 1 54 15 PM

### CODE REVIEW

File: app/values/tile_path.rb [hunk 1]
```diff
@@ -48,13 +48,20 @@ def tileserver
...
       file_count = resource.decorate.try(:mosaic_file_count)
       return false unless file_count&.positive?
-      return false if resource.is_a?(ScannedMap) && resource.decorate.scanned_maps_count <= 1
       true
     end
 
     def mosaic?
-      return true unless resource.is_a?(RasterResource)
-      return true if resource.decorate.decorated_raster_resources.count.positive?
+      return true if resource.is_a?(RasterResource) && resource.decorate.raster_resources_count > 1
```
Comment (hackartisan) [inline app/values/tile_path.rb:55]: I think we'd like to prefer `Wayfinder.for(resource).raster_resources_count` over `resource.decorate.raster_resources_count`

File: app/services/tile_metadata_service.rb [hunk 2]
```diff
@@ -21,8 +21,8 @@ def path
   end
 
   def mosaic?
-    return true unless resource.is_a?(RasterResource)
-    return true if resource.decorate.raster_resources_count.positive?
+    return true if resource.is_a?(RasterResource) && resource.decorate.raster_resources_count.positive?
```
Comment (hackartisan) [inline app/services/tile_metadata_service.rb:24]: Prefer using a wayfinder rather than a decorator here as well

**Review by hackartisan — CHANGES_REQUESTED** [review]
It's not obvious to me why we're checking for positive or `> 1` in any given case. Maybe you could add comments to say why these are the checks we use?

**Comment by eliotjordan** [comment]
@hackartisan I simplified the mosaic method, added some comments, and fixed some logic that was incorrect in our original implementation. A "mosaic" should have two or more raster files.

File: spec/services/tile_metadata_service_spec.rb [hunk 3]
```diff
@@ -63,14 +65,6 @@
         expect(path).to eq(cloud_path.join("33", "1d", "70", "331d70a54bd94a6580e4763c8f6b34fd", "mosaic-#{fingerprint}.json").to_s)
       end
     end
-
-    context "when there aren't any files on the raster members" do
```
Comment (hackartisan) [inline spec/services/tile_metadata_service_spec.rb:67]: Why don't we want a test for this case anymore?

**Review by hackartisan — CHANGES_REQUESTED** [review]
Thank you that is much more clear! I just have a question about this deleted test. Does it does something else now? Should we update it? Not sure why we want to delete it.

File: spec/services/tile_metadata_service_spec.rb (hunk 3 above)
Comment (eliotjordan) [reply spec/services/tile_metadata_service_spec.rb]: Good catch. We do, I just needed to move the `raise Error` elsewhere.

**Review by hackartisan — APPROVED** [review]

**Pull request merged**